### PR TITLE
Add --debug flag that turns on per request logging in the form of {ME…

### PR DESCRIPTION
**Pull request by debus**
…THOD} {PATH} {TIME_TAKEN}

## This PR addresses the following issues: 

https://github.com/ImageWare/TLSential/issues/51

### Context

Currently there is very minimal logging. This aims to add a litte bit

### Approach

This PR adds a flag `--debug` which turns on per-request logging. Each request is first routed to logging middleware that will log `{METHOD} {PATH} {TIME_TAKEN}`.

### Testing

Enabled the flag and went to /ui/dashboard.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -133,6 +143,24 @@ func main() {
...
+	}
+	return handler
+}
+
+func requestLoggingMiddleWare(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		start := time.Now()
+
+		//Defer in case one of the handlers down the line calls panic
+		defer func() { log.Println(r.Method, r.URL.Path, time.Since(start)) }()
```
Comment (brooks42) [inline main.go:158]: Does defer still guarantee order? so if you log the same thing several times is this defer running on the same thread or something so the logs always come out the same?

Comment (d1str0) [reply main.go:158]: defer runs on the same thread.

**Review by d1str0 — APPROVED** [review]
LGTM. I dig the chainMiddleware

File: main.go (hunk 1 above)
Comment (debus) [reply main.go:158]: Yeah defer runs on the same thread. Its sort of like a finally block in Java

**Pull request merged**